src/prop/proof_cnf_stream.h

```diff
@@ -90,6 +91,19 @@ class ProofCnfStream : protected EnvObj
    */
   void getBooleanVariables(std::vector<TNode>& outputVariables) const;
 
+  /**
+   * Dump dimacs of the given clauses to the given output stream.
+   * For details, see cnf_stream.h.
+   */
+  void dumpDimacs(std::ostream& out, const std::vector<Node>& clauses);
+  /**
+   * Same as above, but also prints additional "auxliary unit" clauses.
+   * For details, see cnf_stream.h.
+   */
+  void dumpDimacs(std::ostream& out,
```

Comment:
Why have two versions publicly? You expect any use where you'd not be printing the auxiliary units?

Reply:
Yes, I believe @abdoo8080 mentioned that he had a use case for this method?

Reply:
@limpa105, a new PhD student in our lab, requested this feature for her project. I hacked a branch with Andy's code for her to use at the time. This PR is a step towards removing those hacks (it would be nice to expose it via the API or options).

Reply:
I see, cool!